odxtools/parameters/parameterwithdop.py

```diff
@@ -39,20 +40,20 @@ def dop(self) -> Union[DopBase, None]:
     def dop(self, dop: DopBase):
         self._dop = dop
         if not self.dop_snref:
-            self.dop_ref = dop.id
+            self.dop_ref = OdxLinkRef.from_id(dop.id)
         else:
             self.dop_snref = dop.short_name
 
-    def resolve_references(self, parent_dl, id_lookup):
+    def resolve_references(self, parent_dl, odxlinks):
         dop = self.dop
         if self.dop_snref:
-            dop = parent_dl.data_object_properties[self.dop_snref]
-            if not dop:
+            dop = parent_dl.data_object_properties.get(self.dop_snref)
```

Comment:
get ? data_object_properties is not a dict, it's NamedItemList and does not have a get function